deployment/checkreleases.py

```diff
@@ -0,0 +1,64 @@
+from logging import getLogger
+from datetime import timedelta, datetime
+from threading import Condition
+from itertools import repeat
+import json
+
+from .instancehealth import InstanceHealth
+from . import database, samodels as m
+from . import execution, executils
+
+logger = getLogger(__name__)
+
+
+class CheckReleasesWorker(object):
+    """Check releases from every servers and update health info."""
+
+    def __init__(self, frequency, health):
+        logger.info("CheckReleases worker init. It will run every %d seconds" % frequency)
+        self._running = True
+        self._frequency = frequency
+        self._health = health
+        self.wakeup_period = timedelta(seconds=self._frequency)
+        self.condition = Condition()
+
+    def start(self):
+        logger.info("CheckReleases worker wakeup.")
```

Comment:
The wakeup occurs in the while loop (this line should be below `while self._running:`)